src/ComposerRequireChecker/JsonLoader.php

```diff
@@ -37,6 +41,7 @@ public static function getData(string $path): array
             throw new InvalidJson('error parsing ' . $path . ': ' . $exception->getMessage(), 0, $exception);
         }
 
+        /** @psalm-var ComposerData $decodedData */
```

Comment:
Is any meaningful assertion possible here? Trusting the JSON data to be valid seems a bit extreme.

I usually use @azjezz /psl for that

Reply:
I guess you mean `PSL\Json\typed()`?

i can introduce the dependency if you like.

Reply:
IMO worth adding indeed 👍 

Not sure about whether `phpstan/phpstan` will be happy about it - we'll see :D

Reply:
doc: https://php-standard-library.github.io/#/components/json/

phpstan probably won't be happy :D especially if you are using Type\shape([...]), as that only works properly when using the psl psalm plugin ( https://github.com/php-standard-library/psalm-plugin/blob/main/src/EventHandler/Type/Shape/FunctionReturnTypeProvider.php )

Reply:
Yep, but dropping phpstan is no biggie either

Reply:
If you would like, you can merge this PR as is, and i will send a PR similar to the ones i sent for Roave/* and Laminas/* 😉

Reply:
@dkreuer your choice on that :D

Reply:
Having help in that would be great.

Reply:
cool, you will send a PR for PSL integration this weekend 👍